public/app/features/explore/Logs/Logs.tsx

```diff
@@ -248,84 +239,103 @@ class UnthemedLogs extends PureComponent<Props, State> {
+            color: LogLevelColor[level.logLevel],
+          });
+        }
       });
     }
-  };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [logsVolumeData?.data, hiddenLogLevels]);
 
-  updatePanelState = (logsPanelState: Partial<ExploreLogsPanelState>) => {
-    const state: ExploreItemState | undefined = getState().explore.panes[this.props.exploreId];
-    if (state?.panelsState) {
-      dispatch(
-        changePanelState(this.props.exploreId, 'logs', {
-          ...state.panelsState.logs,
-          columns: logsPanelState.columns ?? this.props.panelState?.logs?.columns,
-          visualisationType: logsPanelState.visualisationType ?? this.state.visualisationType,
-          labelFieldName: logsPanelState.labelFieldName,
-          refId: logsPanelState.refId ?? this.props.panelState?.logs?.refId,
-        })
-      );
-    }
-  };
+  useEffect(() => {
+    return () => {
+      if (flipOrderTimer) {
+        window.clearTimeout(flipOrderTimer);
+      }
 
-  getNumberOfLogVolumes() {
-    const data = this.props.logsVolumeData?.data.filter(
-      (frame: DataFrame) => frame.meta?.dataTopic !== DataTopic.Annotations
-    );
-    const grouped = groupBy(data, 'meta.custom.datasourceName');
-    const numberOfLogVolumes = Object.keys(grouped).length;
-    return numberOfLogVolumes;
-  }
+      if (cancelFlippingTimer) {
+        window.clearTimeout(cancelFlippingTimer);
+      }
 
-  componentDidUpdate(prevProps: Readonly<Props>, prevState: Readonly<State>): void {
-    if (this.props.loading && !prevProps.loading && this.props.panelState?.logs?.id) {
+      // If we're unmounting logs (e.g. switching to another datasource), we need to remove the table specific panel state, otherwise it will persist in the explore url
+      if (panelState?.logs?.columns || panelState?.logs?.refId || panelState?.logs?.labelFieldName) {
+        dispatch(
+          changePanelState(exploreId, 'logs', {
+            ...panelState?.logs,
+            columns: undefined,
+            visualisationType: visualisationType,
+            labelFieldName: undefined,
+            refId: undefined,
+          })
+        );
+      }
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    if (loading && !previousLoading && panelState?.logs?.id) {
       // loading stopped, so we need to remove any permalinked log lines
-      delete this.props.panelState.logs.id;
+      delete panelState.logs.id;
 
       dispatch(
-        changePanelState(this.props.exploreId, 'logs', {
-          ...this.props.panelState,
+        changePanelState(exploreId, 'logs', {
+          ...panelState,
         })
       );
     }
-    if (this.props.panelState?.logs?.visualisationType !== prevProps.panelState?.logs?.visualisationType) {
-      const visualisationType = this.props.panelState?.logs?.visualisationType ?? getDefaultVisualisationType();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Same here. Forcing the dependencies to not update should not be the solution, because the point of the dependency array and the linting rule is to use valid referenced values and prevent bugs.